src/Microsoft.IdentityModel.Logging/WatchUtility.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// A utility class to measure time.
+    /// </summary>
+    internal class WatchUtility
+    {
+        internal static readonly Stopwatch Watch = Stopwatch.StartNew();
```

Comment:
This isn't the way we want to do this, we shouldn't have a static StopWatch always running.

Use a TimeProvider to get the time at the start and end of operations, you can also measure in milliseconds directly and not ticks.

Reply:
Would modifying the use so that it's only running during the interval being measured still be too costly?